node/test/Infrastructure/Generator.hs

```diff
@@ -129,18 +126,19 @@ simpleModel = GeneratorModel {
       gmAllAddresses  = addrs
     , gmEstimateFee   = \_ _ -> 0
     , gmBootstrapStakeholders = Addr <$> [0..3]
-    , gmBoot          = DSL.Transaction {
+    , gmBoot          = Transaction {
                             trFresh = fromIntegral (length addrs) * initBal
-                          , trIns   = Set.empty
-                          , trOuts  = [DSL.Output a initBal | a <- addrs]
+                          , trIns   = error "Boot inputs"
```

Comment:
Does this mean that we do want to to have transactions without empty inputs except in the boot transaction? So I guess the idea here is that if somebody tries to use the `trIns` of a boot transaction the model will blow in her face (signaling an error in the executable spec)?

Don't we want to have transactions with empty inputs?

Reply:
Not according to the rule based spec document, which is what I was following. Whether this is a sensible choice or not, I haven't yet formed an opinion :-)

Reply:
Personally I'd like to avoid having `error`s unless strictly necessary, but maybe this is one of these cases :)

Also the flexibility of specifying empty inputs might be useful, but probably we should plan a WIP pull request so that other people can take a look at it.